Mathlib/Topology/Baire/BaireMeasurable.lean

```diff
@@ -0,0 +1,184 @@
+
+theorem _root_.IsMeagre.baireMeasurableSet (h : IsMeagre s) : BaireMeasurableSet s :=
+  (of_mem_residual h).of_compl
+
+theorem iUnion {ι : Sort*} [Countable ι] {s : ι → Set α}
+    (h : ∀ i, BaireMeasurableSet (s i)) : BaireMeasurableSet (⋃ i, s i) :=
+  MeasurableSet.iUnion h
+
+theorem biUnion {ι : Type*}  {s : ι → Set α} {t : Set ι} (ht : t.Countable)
+    (h : ∀ i ∈ t, BaireMeasurableSet (s i)) : BaireMeasurableSet (⋃ i ∈ t, s i) :=
+  MeasurableSet.biUnion ht h
+
+theorem sUnion {s : Set (Set α)} (hs : s.Countable)
+    (h : ∀ t ∈ s, BaireMeasurableSet t) : BaireMeasurableSet (⋃₀ s) :=
+  MeasurableSet.sUnion hs h
+
+theorem iInter {ι : Sort*} [Countable ι] {s : ι → Set α}
+    (h : ∀ i, BaireMeasurableSet (s i)) : BaireMeasurableSet (⋂ i, s i) :=
+  MeasurableSet.iInter h
+
+theorem biInter {ι : Type*}  {s : ι → Set α} {t : Set ι} (ht : t.Countable)
+    (h : ∀ i ∈ t, BaireMeasurableSet (s i)) : BaireMeasurableSet (⋂ i ∈ t, s i) :=
+  MeasurableSet.biInter ht h
+
+theorem sInter {s : Set (Set α)} (hs : s.Countable)
+    (h : ∀ t ∈ s, BaireMeasurableSet t) : BaireMeasurableSet (⋂₀ s) :=
+  MeasurableSet.sInter hs h
+
+theorem union (hs : BaireMeasurableSet s) (ht : BaireMeasurableSet t) :
+    BaireMeasurableSet (s ∪ t) :=
+  MeasurableSet.union hs ht
+
+theorem inter (hs : BaireMeasurableSet s) (ht : BaireMeasurableSet t) :
+    BaireMeasurableSet (s ∩ t) :=
+  MeasurableSet.inter hs ht
+
+theorem diff (hs : BaireMeasurableSet s) (ht : BaireMeasurableSet t) :
+    BaireMeasurableSet (s \ t) :=
+  MeasurableSet.diff hs ht
+
+theorem congr (hs : BaireMeasurableSet s) (h : s =ᵇ t) : BaireMeasurableSet t :=
+  EventuallyMeasurableSet.congr (α := α) hs h.symm
+
+end BaireMeasurableSet
+
+open Filter
+
+/--Any Borel set differs from some open set by a meager set. -/
+theorem MeasurableSet.residualEq_open [MeasurableSpace α] [BorelSpace α] (h : MeasurableSet s) :
+    ∃ u : Set α, (IsOpen u) ∧ s =ᵇ u := by
+  apply h.induction_on_open (fun s hs => ⟨s, hs, EventuallyEq.rfl⟩)
+  · rintro s - ⟨u, uo, su⟩
+    refine ⟨(closure u)ᶜ, isClosed_closure.isOpen_compl,
+      EventuallyEq.compl (su.trans $ EventuallyLE.antisymm subset_closure.eventuallyLE ?_)⟩
+    have : (u ∪ (closure u)ᶜ) ∈ residual _ :=
+      residual_of_dense_open (uo.union isClosed_closure.isOpen_compl) union_compl_closure_dense
+    rw[EventuallyLE]
+    convert this
+    simp only [le_Prop_eq, imp_iff_or_not]
+    rfl --terrible
+  rintro s - - hs
+  choose u uo su using hs
+  exact ⟨⋃ i, u i, isOpen_iUnion uo, EventuallyEq.countable_iUnion su⟩
+
+/--Any `BaireMeasurableSet` differs from some open set by a meager set. -/
+theorem BaireMeasurableSet.residualEq_open (h : BaireMeasurableSet s) :
+    ∃ u : Set α, (IsOpen u) ∧ s =ᵇ u := by
+  borelize α
+  rcases h with ⟨t, ht, hst⟩
+  rcases ht.residualEq_open with ⟨u, hu, htu⟩
+  exact ⟨u, hu, hst.trans htu⟩
+
+section Map
+
+open Set
+
+variable {f : α → β} (hc : Continuous f) (ho : IsOpenMap f)
```

Comment:
As @sgouezel told me a few years ago, using `variable`s for explicit arguments is not a good style. It's not hard to copy+paste them.